reana_workflow_engine_yadage/utils.py

```diff
@@ -10,13 +10,15 @@
 from reana_commons.publisher import WorkflowStatusPublisher
 
 
-class REANAWorkflowStatusPublisher(object):
+class WorkflowStatusPublisherSingleton(object):
```

Comment:
Is there a benefit of naming the class with ending "...Singleton"?  Just wondering, since we don't do it systematically, e.g. in r-j-controller.  So we could simply end up with "...Publisher"?

Reply:
Since [`WorkflowStatusPublisher`](https://reana-commons.readthedocs.io/en/latest/#reana_commons.publisher.WorkflowStatusPublisher) needs to be accessible globally in Yadage I thought that adding singleton would be more explicit/understandable inside the engine code than calling it `REANAWorkflowStatusPublisher`. I will change it back since it is true it is not harmonised with the rest of the singletons in REANA 👍🏽 .

P.S. Perhaps when implementing other engines we find that the base WorkflowStatusPublisher should be a singleton by default and then we can get rid of the patch here.